### PR TITLE
refactor: improve modal static methods type definition

### PR DESCRIPTION
**Pull request by BANG88**
Discussion #5079

### CODE REVIEW

File: components/modal/Modal.tsx [hunk 1]
```diff
@@ -43,14 +44,17 @@ export interface ModalContext {
     Modal?: any,
   };
 }
+type ModalFunc = (props: ModalFuncProps) => {
```
Comment (infeng) [inline components/modal/Modal.tsx]: add `export`

File: components/modal/Modal.tsx [hunk 2]
```diff
@@ -3,6 +3,7 @@ import React from 'react';
 import Dialog from 'rc-dialog';
 import addEventListener from 'rc-util/lib/Dom/addEventListener';
 import Button from '../button';
+import { ModalFuncProps } from './';
```
Comment (infeng) [inline components/modal/Modal.tsx]: circular reference?

Comment (bang88) [inline components/modal/Modal.tsx]: type definitions looks no problem here . 😄

**Review by infeng — APPROVED** [review]

**Comment by benjycui** [comment]
Due to network problem, I cannot view reason for CI failed.

You can try to rebase first.

**Comment by yesmeck** [comment]
是因为 S3 挂了，导致 travis 也挂了。。。

**Comment by bang88** [comment]
The tests make me crazy...

**Comment by afc163** [comment]
Rebase master.

**Comment by bang88** [comment]
I just rebase it a moment ago

**Comment by afc163** [comment]
I fixed ci in https://github.com/ant-design/ant-design/commit/5eef3fa467302dbee00660a8eb598624b0b78570 a few seconds ago.

**Comment by bang88** [comment]
😢

**Comment by afc163** [comment]
Something wrong with travis.

**Comment by afc163** [comment]
Rebase master again, it finally works in https://travis-ci.org/ant-design/ant-design/builds/206499132

**Comment by bang88** [comment]
Later . no computer this moment...

**Pull request merged**